repos/system_upgrade/common/actors/detectgrubconfigerror/actor.py

```diff
@@ -0,0 +1,73 @@
+            report_fields = [
+                reporting.Title('Syntax error detected in grub configuration'),
+                reporting.Summary(
+                    'Syntax error was detected in GRUB_CMDLINE_LINUX value of grub configuration. '
+                    'This error is causing booting and other issues. '
+                    'Error is automatically fixed by add_upgrade_boot_entry actor.'
+                ),
+                reporting.Severity(reporting.Severity.LOW),
+                reporting.Groups([reporting.Groups.BOOT]),
+            ]
+            report_fields.extend([reporting.RelatedResource('file', config_file) for config_file in error.files])
+            create_report(report_fields)
+        # missing newline, recoverable
+        for error in [err for err in self.consume(GrubConfigError)
+                      if err.error_type == GrubConfigError.ERROR_MISSING_NEWLINE]:
+            report_fields = [
+                reporting.Title('Detected a missing newline at the end of grub configuration file.'),
+                reporting.Summary(
+                    'The missing newline in /etc/default/grub causes booting issues when appending '
+                    'new entries to this file during the upgrade. Leapp will automatically fix this '
+                    'problem by appending the missing newline to the grub configuration file.'
+                ),
+                reporting.Severity(reporting.Severity.LOW),
+                reporting.Groups([reporting.Groups.BOOT]),
+            ]
+            report_fields.extend(reporting.RelatedResource('file', config) for config in error.files)
+            create_report(report_fields)
+        # corrupted configuration, inhibitor
+        for error in [err for err in self.consume(GrubConfigError)
+                      if err.error_type == GrubConfigError.ERROR_CORRUPTED_GRUBENV]:
+            report_fields = [
+                reporting.Title('Detected a corrupted grubenv file.'),
+                reporting.Summary(
+                    'The grubenv file must be valid to pass the upgrade correctly: \n'
+                    '- an exact size of 1024 bytes is expected \n'
+                    '- it cannot end with a newline. \n'
+                    'The corruption could be caused by a manual modification of the file which is not recommended.'
+                ),
+                reporting.Remediation(
+                    hint='Delete {} file(s) and regenerate grubenv using the grub2-mkconfig tool'.format(
+                        ','.join(error.files))),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Groups([reporting.Groups.BOOT]),
+                reporting.Groups([reporting.Groups.INHIBITOR]),
+            ]
+            report_fields.extend([reporting.RelatedResource('file', config) for config in error.files])
+            create_report(report_fields)
```

Comment:
@fernflower  can you actually put it into the library and split it into functions? also it will be better to nest the `if` statements into the for-loop  to improve the readability. Also I would add `break` after each report (or return when you put it into functions) so it's clear that it's not possible to see multiple reports for the same thing

Reply:
@pirat89 IMHO splitting this into functions brings a little in terms of value. It might be me getting old and grumpy, but unless the code is called from multiple places, or the logic of what is being done in a block is very complex, I prefer linear stream of code, because it is just easier to debug when you don't have to deal with super-deep call stacks. Also, splitting this into several functions does not increase testability, as the tests can be essentially the same if it is one big function of several smaller. (_If an error of type X was detected, is inhibitor generated?_ - only the tested function is changed)

Reply:
But I agree that on that having one forloop with if-elif-else branches inside would be more readable in a sense that it is obvious that any error can be reported precisely once.

Reply:
@fernflower  @MichalHe I do not have strong opinion here - definitely not blocker at all. This is more about preferences. If you guys find it better in this way, I am ok with it. I thought it would be better/more readable to decompose it into smaller pieces, and adding breaks could be also seatbelts, so we are sure it will not happen in future that some reports appear multiple times (e.g. if an additional grub error msg is produced in future that is not expected now). But I do not have problem to keep it is if others find it better in this way. Thanks both for reacting :-)